packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -53,29 +53,20 @@ impl<C: AsClient> PostgresStore<C> {
             let dependency_status = dependency_context
                 .knowledge_dependency_map
                 .insert(&entity_edition_id, current_resolve_depth);
-            let entity: Option<&KnowledgeGraphVertex> = match dependency_status {
-                DependencyStatus::Unknown => {
-                    if let Some(entity) = subgraph.vertices.knowledge_graph.get(&entity_edition_id)
-                    {
-                        Some(entity)
-                    } else {
-                        let entity = Read::<Entity>::read_one(
-                            self,
-                            &Filter::for_entity_by_edition_id(entity_edition_id),
-                        )
-                        .await?;
-                        Some(
-                            subgraph
-                                .vertices
-                                .knowledge_graph
-                                .entry(entity_edition_id)
-                                .or_insert(KnowledgeGraphVertex::Entity(entity)),
-                        )
+            let entity = match dependency_status {
+                DependencyStatus::Unresolved => {
+                    match subgraph.vertices.knowledge_graph.entry(entity_edition_id) {
+                        Entry::Occupied(entry) => Some(entry.into_mut()),
+                        Entry::Vacant(entry) => {
+                            let entity = Read::<Entity>::read_one(
+                                self,
+                                &Filter::<Entity>::for_entity_by_edition_id(entity_edition_id),
+                            )
+                            .await?;
+                            Some(entry.insert(KnowledgeGraphVertex::Entity(entity)))
```

Comment:
I had a quick dig through the API but didn't find anything nice to do this, but do we actually want mutable references here? I think it could be misleading

```suggestion
                        Entry::Occupied(entry) => Some(&*entry.into_mut()),
                        Entry::Vacant(entry) => {
                            let entity = Read::<Entity>::read_one(
                                self,
                                &Filter::<Entity>::for_entity_by_edition_id(entity_edition_id),
                            )
                            .await?;
                            Some(&*entry.insert(KnowledgeGraphVertex::Entity(entity)))
```

Ideally there's a way for us to avoid the mutable reference in the first place

Reply:
Do we maybe want a small comment saying something like:
```rust
// Explicit typing means we don't take mutable references when these records shouldn't be mutable
```

Reply:
Good point, I added a comment in a5d8d95d344ea9cb0c9648cbd8f9880db8dc071e